src/pudl/transform/ferc1.py

```diff
@@ -1370,9 +1383,55 @@ def apply_sign_conventions(self, df) -> pd.DataFrame:
 
         return df
 
+    def targeted_drop_duplicates_dbf(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Drop bad duplicate records from a specific utility in 2018.
+
+        This is a very specific fix, meant to get rid of a particular observed set of
+        duplicate records: FERC Respondent ID 187 in 2018 has two sets of plant in
+        service records, one of which contains a bunch of null data.
+
+        This method is part of the DBF processing because we want to be able to
+        hard-code a specific value of ``utility_id_ferc1_dbf`` and those IDs are no
+        longer available later in the process. I think.
```

Comment:
i think this location makes sense. this is a very table specific transform rn so it should definitely be held within this class. It technically could be a function instead of a method (except it uses `self.table_id`...), but with our current architecture I think this is fine.

I could imagine running it during many steps of the transform process but i think doing it early in `process_dbf` makes sense. The only thing that makes me question that is that this method is also checking for duplicates overall, which could be a good thing to do during `transform_main` instead of doing it in `process_dbf`.

Reply:
Hmm yeah I guess you're right that's too general of a check for this step.

Actually it would be **really nice** to check every table for duplicates based on PK at the end of the transformation, so we hear about it before we get down the road to the DB load step. Maybe I should add this to `transform_end()`?

Reply:
great idea but this is a *later* check imo